src/main/java/org/folio/rest/impl/PostOrdersHelper.java

```diff
@@ -133,4 +141,11 @@ public Void handleError(Throwable throwable) {
 
     return null;
   }
+
+  public static String generatePoNumber() {
+    return (Long.toHexString(System.currentTimeMillis() - PO_NUMBER_EPOCH) +
+        Long.toHexString(System.nanoTime() % 100))
+          .toUpperCase();
+  }
```

Comment:
This algorithm works, no need to account for race conditions, but a different algorithm for a particular naming scheme may require race conditions management.